tests/testthat/test-paginate_listing.R

```diff
@@ -44,3 +44,62 @@ testthat::test_that("pagination works vertically", {
                          page6_expected)
 
 })
+
+testthat::test_that("horizontal pagination with 0 or 1 key column specified works correctly", {
+  # pre-processing and ordering
+  tmp_data <- ex_adae %>%
+    dplyr::slice(1:30) %>%
+    distinct(USUBJID, AGE, BMRKR1, .keep_all = TRUE)
+
+  lsting <- as_listing(tmp_data,
+    key_cols = c("USUBJID")
+  ) %>%
+    add_listing_col("AGE") %>%
+    add_listing_col("BMRKR1", format = "xx.x") %>%
+    add_listing_col("BMRKR2")
+
+  pages_listings <- paginate_listing(lsting, cpp = 70, verbose = TRUE)
+  pg1_header <- strsplit(toString(matrix_form(pages_listings[[1]])), "\n")[[1]][1:2]
+  pg2_header <- strsplit(toString(matrix_form(pages_listings[[2]])), "\n")[[1]][1:2]
+  pg1_header_expected <- c(
+    "Unique Subject Identifier   Age   Continous Level Biomarker 1",
+    "—————————————————————————————————————————————————————————————"
+  )
+  pg2_header_expected <- c(
+    "Unique Subject Identifier   Categorical Level Biomarker 2",
+    "—————————————————————————————————————————————————————————"
+  )
+
+  testthat::expect_equal(pg1_header, pg1_header_expected)
+  testthat::expect_equal(pg2_header, pg2_header_expected)
+  testthat::expect_equal(length(pages_listings), 2L)
+
+  lsting2 <- as_listing(tmp_data) %>%
```

Comment:
FYI, This actually has 1 key col; we can tell by the fact that Study Identifier is the first column (label) on all the pages. The default key_cols value (currently) is to take the first column. 

Thats ok though because under local changes I have and will be pushing pretty soon, actually specifying 0 key columns is throws an error, so I will fix that on my side and roll it in with my changes.